lib/tpcal.py

```diff
@@ -1,33 +1,34 @@
+
 import lvgl as lv
 lv.init()
 
-lv.log_register_print_cb(lambda level,path,line,msg: print('LOG: %s(%d): %s' % (path, line, msg)))
-
 # Initialize ILI9341 display
 
 import lvesp32
-from ili9XXX import ili9341
-disp = ili9341()
-
-HRES = lv.disp_get_hor_res(lv.disp_t.cast(None))
-VRES = lv.disp_get_ver_res(lv.disp_t.cast(None))
-
-# Register raw resistive touch driver
-'''
-import rtch
-touch = rtch.touch(xp = 32, yp = 33, xm = 25, ym = 26, touch_rail = 27, touch_sense = 33, cal_x0=0, cal_x1 = HRES, cal_y0=0, cal_y1 = VRES)
-touch.init()
-indev_drv = lv.indev_drv_t()
-lv.indev_drv_init(indev_drv) 
-indev_drv.type = lv.INDEV_TYPE.POINTER;
-indev_drv.read_cb = touch.read;
-lv.indev_drv_register(indev_drv);
-'''
+import espidf as esp
+from ili9XXX import ili9341,COLOR_MODE_BGR,LANDSCAPE,PORTRAIT
+
+# select the orientation
+# orientation = LANDSCAPE
+orientation = PORTRAIT
+w=320
+h=240
```

Comment:
It is hard coded anyway. Either in tpcal or in the width and height parameters in the ili9XXX driver initialization.
I defined it in tpcal because like this I can easily switch width and height when switching from portrait to landscape mode.

Reply:
You can extract height/width from LVGL in a number of ways.
For example:
```python
w = lv.scr_act().get_width()
h = lv.scr_act().get_height()
```
Another example:
```python
w = lv.disp_get_default().get_hor_res()
h = lv.disp_get_default().get_ver_res()
```

In case of the current ili9341, it's true that you can provide them to the driver during initialization. But on other drivers it might not even be configurable.  
Actually it makes more sense to rely on ili9341 default values, and change the ili9XXX driver to switch width and height according to `rot` parameter. This way you don't have to explicitly specify width/height on your script at all.

When it's hard coded, if someone ever replaces the display driver on this script, he would have to remember to replace the hard coded values as well.
 
So I suggest deriving width and height from LVGL instead of hard coding them, and possibly avoid explicitly specifying the width/height values at all in this script.

Reply:
The problem only comes from the fact that this program allows calibration in landscape **and** portrait mode. The v6 program only did the calibration in default (portrait) orientation.
I now got rid of the problem going back to the v6 strategy. The program now calibrates in portrait mode only. However, I give an indication what needs to be done for landscape calibration in the comments.
I also get back to driver initialization with default parameters and give the parameters for the board I have only in comments. I believe that the program will run ok but cannot test it because of the different hardware I have.
Is this acceptable?